usr.sbin/gssd/gssd.c

```diff
@@ -202,6 +202,7 @@ main(int argc, char **argv)
 		signal(SIGHUP, SIG_IGN);
 	}
 	signal(SIGTERM, gssd_terminate);
+	signal(SIGPIPE, gssd_terminate);
```

Comment:
Should it terminate on SIGPIPE, or just ignore and handle the respective error?  I don't know what pipes it uses, but for daemon I'd expect not dying on first error, even cleanly, unless the error is really crucial.

Reply:
The current behaviour is that it dies.  The FD in question is the socket that it passes down to the kernel, which for an as-yet-undiagnosed reason closes it.  Since the daemon doesn't have much in the way of recovery code, dying cleanly seems the best workaround for now.